modules/sfp_ipqualityscore.py

```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
         fraudScore = data.get('fraud_score')
         recentAbuse = data.get('recent_abuse')
         botStatus = data.get('bot_status')
+        malicious = False
+        maliciousDesc = ""
+
         if fraudScore >= self.opts['abuse_score_threshold'] or recentAbuse or botStatus:
             evt = SpiderFootEvent("RAW_RIR_DATA", str(data), self.__name__, event)
             self.notifyListeners(evt)
+            malicious = True
+            maliciousDesc = f"IPQualityScore [ {eventData} ]\n"
+
+        if eventName == "PHONE_NUMBER":
+            if malicious:
+                maliciousDesc += f" - FRAUD SCORE : {data.get('fraud_score')}\n - ACTIVE : {data.get('active')}\n - RISKY : {data.get('risky')}\n - RECENT ABUSE : {data.get('recent_abuse')}"
+                evt = SpiderFootEvent("MALICIOUS_PHONE_NUMBER", maliciousDesc, self.__name__, event)
+                self.notifyListeners(evt)
 
-            if eventName == "PHONE_NUMBER":
-                evt = SpiderFootEvent("MALICIOUS_PHONE_NUMBER", "IPQualityScore [" + eventData + "]", self.__name__, event)
+            geoInfo = ""
+            city = data.get('city')
+            country = data.get('country')
+            zipcode = data.get('zip_code')
+            region = data.get('region')
+
+            if city:
+                geoInfo += city + ", "
+            if region:
+                geoInfo += region + ", "
+            if country:
+                geoInfo += country + " "
+            if zipcode:
+                geoInfo += zipcode + ". "
+
+            evt = SpiderFootEvent("GEOINFO", geoInfo, self.__name__, event)
```

Comment:
`geoinfo` may still be empty by this stage, so you should wrap this in a `if geoInfo` clause.